stdlib/obj.mli

```diff
@@ -55,6 +55,7 @@ external field : t -> int -> t = "%obj_field"
 *)
 external set_field : t -> int -> t -> unit = "%obj_set_field"
 external set_tag : t -> int -> unit = "caml_obj_set_tag"
+  [@@ocaml.deprecated "Use with_tag instead."]
```

Comment:
It's not necessarily a great idea to deprecate a feature at the same version that the replacement is added: a maintainer that would follow the deprecation advice must force its users to upgrade to the latest version (or use conditionals). Could we wait for one version between `with_tag` and `set_tag`'s deprecation?

Reply:
Happy to follow whatever deprecation policy works best here, but I don't really understand the issue. I thought `@@ocaml.deprecated` was the way to signal that a replacement is available? Is the concern here programs compiled with `-warn-error` breaking?

Reply:
Normally, we would deprecate it now and remove it in the next deprecation purge, in about 5 years. We still don't have any mechanism to indicate degrees of deprecation (or how much time before it gets removed). Such a mechanism might be an intermediate step before deprecation.

In the case of `Obj.set_tag` I think we'll remove it out-of-cycle to avoid delaying multicore, so it's better to deprecate right now anyway.

Reply:
> It's not necessarily a great idea to deprecate a feature at the same version that the replacement is added:

Regarding this comment about @gasche I fear a bit nothing is ever going to be deprecated (people will simply forget) if nothing special is done. One simple measure would be to have in the repo text file with a table of the form:

```
ID | vReplacement | vDeprecated notice | vRemove
--------------------------------------------------
```
Before a release one could then apply appropriate changes according to what the table says.

Reply:
@dbuenzli yes, that sounds like a good idea. I guess some of your merged PRs have some maybe-deprecate-later identifiers in flight, do you have enough to send a PR to populate that file? (Did we merge PrintExc successor yet?)

(The file should also point to the (M/G)PR that contains the deprecation action/discussion.)

Reply:
>  (Did we merge PrintExc successor yet?)

No, it is #2137, but it is not yet ready, more work needed.

Reply:
I could have a look but a bit busy® right now®.